boot/bootutil/src/bootutil_public.c

```diff
@@ -316,14 +316,30 @@ int
 boot_write_magic(const struct flash_area *fap)
 {
     uint32_t off;
+    uint32_t pad_off;
     int rc;
+    uint8_t magic[BOOT_MAGIC_ALIGN_SIZE];
+    uint8_t erased_val;   
 
     off = boot_magic_off(fap);
+    /* image_trailer structure was modified with additional padding such that 
+     * the pad+magic ends up in a flash minimum write region. The address 
+     * returned by boot_magic_off() is the start of magic which is not the 
+     * start of the flash write boundary and thus writes to the magic will fail. 
+     * To account for this change, write to magic is first padded with 0xFF 
+     * before writing to the trailer. */
```

Comment:
```suggestion

    /* image_trailer structure was modified with additional padding such that
     * the pad+magic ends up in a flash minimum write region. The address
     * returned by boot_magic_off() is the start of magic which is not the
     * start of the flash write boundary and thus writes to the magic will fail.
     * To account for this change, write to magic is first padded with 0xFF
     * before writing to the trailer.
     */
```
1) Missing blank line before the beginning of the block comment.
2) Dangling whitespaces at the end of each line.
3) For consistency, the terminator for the block comment should be placed in a new line.